src/docfx/build/page/BuildPage.cs

```diff
@@ -64,52 +67,43 @@ public static async Task<List<Error>> Build(Context context, Document file)
             return errors;
         }
 
-        private static async Task<(List<Error> errors, object output, JObject metadata)>
-            CreatePageOutput(Context context, Document file, JObject sourceModel)
+        private static (object output, JObject metadata) CreatePageOutput(Context context, Document file, JObject sourceModel, JObject mergedMetadata)
         {
-            var errors = new List<Error>();
-            var (inputMetaErrors, inputMetadata) = context.MetadataProvider.GetMetadata(file);
-            errors.AddRange(inputMetaErrors);
-
-            var (outputMetaErrors, outputMetadata) = await CreateOutputMetadata(context, file, inputMetadata);
-            errors.AddRange(outputMetaErrors);
-
-            var mergedMetadata = new JObject();
-            JsonUtility.Merge(mergedMetadata, inputMetadata.RawJObject);
-            JsonUtility.Merge(mergedMetadata, JsonUtility.ToJObject(outputMetadata));
-
             var pageModel = new JObject();
             JsonUtility.Merge(pageModel, sourceModel);
             JsonUtility.Merge(pageModel, mergedMetadata);
 
             if (file.Docset.Config.Output.Json && !file.Docset.Legacy)
             {
-                return (errors, pageModel, SortProperties(mergedMetadata));
+                return (pageModel, SortProperties(mergedMetadata));
             }
 
             var (templateModel, metadata) = CreateTemplateModel(context, SortProperties(pageModel), file);
             if (file.Docset.Config.Output.Json)
             {
-                return (errors, templateModel, SortProperties(metadata));
+                return (templateModel, SortProperties(metadata));
             }
 
             var html = context.TemplateEngine.RunLiquid(file, templateModel);
-            return (errors, html, SortProperties(metadata));
+            return (html, SortProperties(metadata));
 
             JObject SortProperties(JObject obj)
                 => new JObject(obj.Properties().OrderBy(p => p.Name));
         }
 
-        private static (List<Error> errors, object output, JObject metadata)
-            CreateDataOutput(Context context, Document file, JObject sourceModel)
-            => (new List<Error>(), context.TemplateEngine.RunJint($"{file.Mime}.json.js", sourceModel), null);
+        private static (object output, JObject metadata)
+            CreateDataOutput(Context context, Document file, JObject sourceModel, JObject mergedMetadata)
+        {
+            sourceModel["metadata"] = mergedMetadata;
```

Comment:
everything will be put into metadata excepted transformedToken?

Reply:
Yes according to Adam